blog/2025-11-12-best-practises-for-writing-promises/index.mdx

```diff
@@ -0,0 +1,559 @@
+```
+
+We can now install this early iteration of the Promise!
+
+```bash
+kubectl apply --filename promise.yaml
+```
+
+You will soon see the Redis dependencies appearing on your worker cluster. Watch the the operator pod as it is created with:
+
+```bash
+kubectl get pods --context kind-worker --watch
+```
+
+You should eventually see output similar to the following:
+
+```bash
+NAME                             READY   STATUS    RESTARTS   AGE
+redisoperator-7f584c6969-zc8sx   1/1     Running   0          41s
+```
+
+You now have a Promise that installs the Redis operator and its its dependencies but it does not yet provision Redis as-a-service
+
+**Declarative Workflows**
+
+Whilst it would have been possible to install the operator and it’s dependencies with `kubectl apply` commands targeting the cluster,
+we have chosen a *declarative* approach; defining the desired state of the resources on the cluster and allowing the configured
+GitOps tools to converge on this desired state. Taking a declarative approach improves reliability as GitOps tools automatically
+correct drift, provide audibility as the configurations are declared up-front, and are more predictable. 
+
+This is also essential
+for ensuring that Kratix can easily delete the resources created when Promises are deleted as Kratix deletes the files created as part of any configure workflows during Promise deletion.
+
+At the moment, our Promise configure workflow is declarative and defines all prerequisites for installing the Redis Operator on
+the cluster, including the CRD for the RedisFailover CR. With these installed, we can continue to to take a declarative approach
+with our resource configure workflows - that is, what happens when a user makes a request of our redis promise.
+
+Within this workflow, we want to create a RedisFailover CR from the inputs provided by the user. There are a number of options
+that can be configured in the RedisFailover CR but there are some defaults that we, as the Promise writers, want to enforce. The
+only thing we want to users to be able to configure is the number of replicas and the name of their RedisFailover. 
+
+We’ll take the name of the RedisFailover from the name of the user’s requests and the number of replicas from the API exposed by
+the Promise. We’re going to wrap the number of replicas in a simple abstraction where `small` is one replica and `large` is three;
+we can implement the logic for this in our workflow which we’ll introduce shortly. 
+
+First, we’re going to update the Promise API to introduce the `size` property. We can do this with the `update api` command:
+
+```bash
+kratix update api --property size:string
+```
+
+You should see that the following has been added to the `promise.yaml`
+
+```yaml
+  api:
+    apiVersion: apiextensions.k8s.io/v1
+    kind: CustomResourceDefinition
+    metadata:
+      creationTimestamp: null
+      name: rediss.mygroup.org
```

Comment:
It's interesting that our command does this, because I'd expect the plural for Redis to be something like "Redises" or something like it.